### PR TITLE
Add 50th and 95th percentiles

### PR DESCRIPTION
**Pull request by mvdan**
These are some of the stats that will later be needed for kubelet to expose via an endpoint.

### CODE REVIEW

File: summary/percentiles.go [hunk 1]
```diff
@@ -88,26 +91,31 @@ func (self *resource) Add(p info.Percentiles) {
...
+		Max:        val,
+		Fifty:      val,
+		Ninety:     val,
+		NinetyFive: val,
 	}
 	self.Add(sample)
 }
 
 // Get max, average, and 90p from existing samples.
-func (self *resource) GetPercentile() info.Percentiles {
```
Comment (rjnagal) [inline summary/percentiles.go]: optional nit: Can we rename to GetAllPercentiles() instead. It makes searching/reading names easier if they don't differ in just the plural.

**Comment by rjnagal** [comment]
LGTM with a small nit.

I think this changes is good for cAdvisor in general.
I haven't yet found a good concise way to put these numbers up on cAdvisor UI (plain table is ugly).
Take a stab at it if you get some time, @mvdan

File: summary/percentiles.go (hunk 1 above)
Comment (mvdan) [inline summary/percentiles.go]: Sure.

**Comment by mvdan** [comment]
Initially these were going to be used by kubelet, but in the end we're doing them in heapster with @afein. If this is in general useful, like you say, it can of course be merged. If not, I'm fine with closing it too.

I can have a look at the UI part, yeah. Should that be on this PR?

**Comment by mvdan** [comment]
nit fixed. Also, note the TODO added. I was just confused by that code and the 90p comment so was hoping someone could shed some light on that.

**Comment by rjnagal** [comment]
The UI comment was an aside, it can be done separately.

We currently get 90p for an hour by using 90p values from last 60 mins sample. If we do the same logic for 95p and 50p, we'll burn a it more memory. I think for now, we only collect up to 60 mins samples, so its not that bad. If we did a day worth of samples per container, then it would become noticeable.

**Comment by mvdan** [comment]
So should I just remove the TODO on the samples line and this can be merged?

**Comment by mvdan** [comment]
@rjnagal Removed the TODO

**Comment by mvdan** [comment]
CC @vishh

**Comment by mvdan** [comment]
I've improved the `GetPercentile(float64)` tests and exported the type.

**Comment by vmarmol** [comment]
ok to test

**Comment by vmarmol** [comment]
LGTM

**Pull request merged**